File: library/src/main/java/sk/vander/library/ActivityModule.java
package sk.vander.library;

import android.app.Activity;
import android.content.Context;

import dagger.Provides;
import sk.vander.library.annotation.ActivityScope;

/**
 * Created by arashid on 26/06/16.
 */
@dagger.Module
@ActivityScope
public class ActivityModule {
  private final Activity activity;

  public ActivityModule(Activity activity) {
    this.activity = activity;
  }

  @Provides @ActivityScope Context provideContext() {
    return activity;
  }
}

File: library/src/main/java/sk/vander/library/ActivityModule.kt
package sk.vander.library

import android.app.Activity
import android.content.Context

import dagger.Provides
import sk.vander.library.annotation.ActivityScope

/**
 * Created by arashid on 26/06/16.
 */
@dagger.Module
@ActivityScope
class ActivityModule(private val activity: Activity) {

  @Provides @ActivityScope fun provideContext(): Context {
    return activity
  }
}
